constraint-solver/src/solver/exhaustive_search.rs

```diff
@@ -171,18 +189,33 @@ fn derive_more_assignments<T: FieldElement, V: Clone + Hash + Ord + Eq + Display
         .flatten()
         .filter_map(|effect| {
             if let Effect::Assignment(variable, value) = effect {
-                Some((variable, value))
+                Some((variable, RangeConstraint::from_value(value)))
+            } else if let Effect::RangeConstraint(variable, rc) = effect {
+                Some((variable, rc))
             } else {
                 None
             }
         })
-        .chain(assignments)
+        .chain(
+            assignments
+                .into_iter()
+                .map(|(v, val)| (v, RangeConstraint::from_value(val))),
+        )
         // Union of all unique assignments, but returning an error if there are any contradictions.
-        .try_fold(BTreeMap::new(), |mut map, (variable, value)| {
-            if let Some(existing) = map.insert(variable, value) {
-                if existing != value {
-                    // Duplicate assignment with different value.
-                    return Err(ContradictingConstraintError);
+        // TODO update
```

Comment:
I don't understand the TODOs. Do you want to fix them before merging? If not, please expand.